py3/SciServer/Dask.py

```diff
@@ -0,0 +1,62 @@
+import requests
+from requests.exceptions import HTTPError
+from base64 import b64decode
+from pathlib import Path
+from dask.distributed import Client
+from distributed.security import Security
+from os.path import expanduser
+import SciServer.Authentication
+import json
+
+def getClient(compute_url=None, ref_id=None):
+    """
+    Creates a new client for the specified Dask cluster.
+    
+    If `compute_url` and `ref_id` are not set, cluster connection properties will be read from the ``~/dask-cluster.json`` file injected into new SciServer Compute containers automatically when they are created with an attached Dask cluster.
+    
+    :param compute_url: the URL of a SciServer Compute instance used to manage Dask clusters, defaults to `None`
+    :param ref_id: Dask cluster reference ID in SciServer Compute, defaults to `None`
+    
+    :return: Dask client object
+    :rtype: :class:`dask.distributed.Client`
+    """
+    token = SciServer.Authentication.getToken()
+    data = None
+    if (compute_url is None):
+        with open(expanduser('~/dask-cluster.json')) as f:
+            data = json.load(f)
+    else:
+        try:
+            response = requests.get(''.join([compute_url.rstrip('/'), '/api/dask/clusters/', ref_id]),
+                                    params = {'connectionInfo': 'true'},
+                                    headers = {'X-Auth-Token': token})
+            response.raise_for_status()
+            data = response.json()['connection']
+        except HTTPError as err:
+            try:
+                msg = str(err) + '\n' + response.json()['error']['stackTrace']
+            except:
+                raise err
+            raise HTTPError(msg)
+    
+    base_dir = expanduser('~/.sciserver/dask')
+    Path(base_dir).mkdir(parents=True, exist_ok=True)
+    
+    with open(base_dir + '/ca.pem', 'w+b') as f:
+        f.write(b64decode(data['ca']))
```

Comment:
I'm a bit confused by the combination of PEM file with the binary file option and base64 decoding. The PEM is a plain ASCII file with the certificate/key contents base64 encoded.

Reply:
That's because Compute encodes the PEM files in base64 before putting them in JSON fields. Maybe not strictly necessary, but we get a nice long string with no line breaks, always nice to wrap files in base64, even if they are supposed to be ascii... Similarly, for example, in k8s PEM files would be encoded as base64 if they were put in a secret.

Reply:
ok, this makes sense. The `+b` would still cause some confusion though, I think

Reply:
Why? b64decode returns bytes.

Reply:
but those bytes ought to be ASCII, in any case, I'm sure it would not make a material difference

Reply:
I don't quite understand... Simply removing `b` would result in an error, obviously. To make it work in text mode I would have to do something like this (unless I'm missing something obvious):
```
with open(base_dir + '/ca.pem', 'w+',encoding='ascii') as f:
        f.write(b64decode(data['ca']).decode('ascii'))
```
This looks more, not less confusing to me... Wouldn't working with bytes be more generic and less error prone?

Reply:
I guess my point is that since that we expect that to be ascii data and not binary, that doing 

```python
with open(base_dir + '/ca.pem', 'w') as f:
    f.write(b64decode(data['ca']).decode('ascii'))
```
Is a bit more explicit, and would fail fast if the data were in fact not encoded as such - rather than down the line when dask workers are making a connection. 

of course it is fine as-is, so I already hit the approval button :)